### PR TITLE
test: upgrade golangci/golangci-lint-action to v1.54

### PR DESCRIPTION
**Pull request by umagnus**
**What type of PR is this?**

/kind test

**What this PR does / why we need it**:

test: upgrade golangci/golangci-lint-action to v1.54

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+linters-settings:
```
Comment (andyzhangx) [inline .golangci.yml:1]: what is this?

Comment (umagnus) [reply .golangci.yml:1]: It‘s a configure for depguard. reference for https://github.com/golangci/golangci-lint/issues/3877#issuecomment-1573682848
The default rule of depguard is to allow only std lib in all files.

File: test/utils/azure/azure_helpers.go [hunk 2]
```diff
@@ -313,7 +313,7 @@ func getCloudConfig(env azure.Environment) cloud.Configuration {
 	}
 }
 
-func getClient(env azure.Environment, subscriptionID, tenantID string, cred *azidentity.ClientSecretCredential, scope string) *Client {
+func getClient(env azure.Environment, subscriptionID, _ string, cred *azidentity.ClientSecretCredential, scope string) *Client {
```
Comment (andyzhangx) [inline test/utils/azure/azure_helpers.go:316]: if a parameter not used in func, then just remove this param in the function

Comment (umagnus) [reply test/utils/azure/azure_helpers.go]: done

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**